sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/Providers.java

```diff
@@ -108,11 +110,11 @@ public TInstance create(Function<TProvider, TInstance> createInstance, TInstance
             provider = defaultProvider;
 
             if (provider == null) {
-                if (fallbackInstance == null) {
+                if (fallbackSupplier == null || fallbackSupplier.get() == null) {
```

Comment:
null check the supplier then if not null call `get` storing the value then null check that. If this isn't returning something shared this will create multiple instances.